examples/webgpu_tsl_interoperability.html

```diff
@@ -87,69 +87,66 @@
-            tex_sampler: sampler,
-            crtWidth: f32, 
-            crtHeight: f32,
-            cellOffset: f32,
-            cellSize: f32, 
-            borderMask: f32,
-            time: f32,
-            pulseIntensity: f32,
-            pulseWidth: f32,
-            pulseRate: f32
-          ) -> vec3<f32> {
-
-            // Convert uv into map of pixels
-            var pixel = (vUv * 0.5 + 0.5) * vec2<f32>(
-              crtWidth, 
-              crtHeight
-            );
-
-            // Coordinate for each cell in the pixel map
-            var coord = pixel / cellSize;
-            // Three color values for each cell (r, g, b)
-            var subcoord = coord * vec2f(3.0, 1.0);
-
-            var cell_offset = vec2<f32>(0, fract(floor(coord.x) * cellOffset));
-
-            var mask_coord = floor(coord + cell_offset) * cellSize;
-
-            var samplePoint = mask_coord / vec2<f32>(crtWidth, crtHeight);
-            samplePoint.x += fract(time / 20);
-
-            var color = textureSample(
-              tex,
-              tex_sampler,
-              samplePoint
-            ).xyz;
-
-            //current implementation does not give an even amount of space to each r, g, b unit of a cell
-            //Fix/hack this by multiplying subCoord.x by cellSize at cellSizes below 6
-            var ind = floor(subcoord.x) % 3;
-
-            var mask_color = vec3<f32>(
-              f32(ind == 0.0), 
-              f32(ind == 1.0), 
-              f32(ind == 2.0)
-            ) * 3.0;
-            
-            var cell_uv = fract(subcoord + cell_offset) * 2.0 - 1.0;
-            var border: vec2<f32> = 1.0 - cell_uv * cell_uv * borderMask;
-            
-            mask_color *= vec3f(clamp(border.x, 0.0, 1.0) * clamp(border.y, 0.0, 1.0));
-            
-            color *= mask_color;
-
-            color.r *= 1.0 + pulseIntensity * sin(pixel.y / pulseWidth + time * pulseRate);
-            color.b *= 1.0 + pulseIntensity * sin(pixel.y / pulseWidth + time * pulseRate);
-            color.g *= 1.0 + pulseIntensity * sin(pixel.y / pulseWidth + time * pulseRate);
-
-            return color;
-          }
-        ` );
+		      fn crtFragment(
+			      vUv: vec2f,
+			      tex: texture_2d<f32>,
+			      texSampler: sampler,
+			      crtWidth: f32, 
+			      crtHeight: f32,
+			      cellOffset: f32,
+			      cellSize: f32,
+			      borderMask: f32,
+			      time: f32,
+			      pulseIntensity: f32,
+			      pulseWidth: f32,
+			      pulseRate: f32
+		      ) -> vec3<f32> {
+			      // Convert uv into map of pixels
+			      var pixel = (vUv * 0.5 + 0.5) * vec2<f32>(
+				      crtWidth, 
+				      crtHeight
+			      );
+			      // Coordinate for each cell in the pixel map
+			      var coord = pixel / cellSize;
+			      // Three color values for each cell (r, g, b)
+			      var subcoord = coord * vec2f(3.0, 1.0);
+			      var offset = vec2<f32>(0, fract(floor(coord.x) * offset));
```

Comment:
Seems that your example broken here after your last revision.

Reply:
I added `cellOffset` but it looks like a black horizontal line in the center of the screen.

Reply:
The black horizontal line is the intended behavior. There are technically two shaders in this pull request, a wgsl Shader on top, and a tsl shader on the bottom. However, they each use the same nodes as their inputs (thus the title, WGSL TSL Node Interoperability).